libs/vkd3d/resource.c

```diff
@@ -6704,3 +7151,75 @@ HRESULT vkd3d_memory_info_init(struct vkd3d_memory_info *info,
           info->cpu_accessible_domain.rt_ds_type_mask);
     return S_OK;
 }
+
+HRESULT vkd3d_global_descriptor_buffer_init(struct vkd3d_global_descriptor_buffer *global_descriptor_buffer,
+        struct d3d12_device *device)
+{
+    VkBufferUsageFlags vk_usage_flags;
+    HRESULT hr;
+
+    /* Don't bother with descriptor buffers if we need to keep offset buffer around. */
+    if (!device->device_info.descriptor_buffer_features.descriptorBuffer ||
+            !device->device_info.descriptor_buffer_features.descriptorBufferPushDescriptors ||
+            device->device_info.properties2.properties.limits.minStorageBufferOffsetAlignment > 4)
```

Comment:
Tried testing Turnip's implementation of `VK_EXT_descriptor_buffer` with vkd3d, but Turnip has `minStorageBufferOffsetAlignment == 64` so we don't pass this check (unfortunately it's a hardware limit, in a descriptor the buffer address is encoded without 5 lowest bits).

> /* Don't bother with descriptor buffers if we need to keep offset buffer around. */

So the high alignment values are worked around by having a separate buffer with per-descriptor offsets, and keeping the offsets buffer around isn't worth due to perf, code complexity, or both?

Reply:
64?!?!?! Dear lord. How would this work on DX12 then? I think it would "just work" to use offset buffers here, it's just not tested.

Reply:
If you can remove the check and stuff is running, we can remove the check.

Reply:
With  `VK_EXT_descriptor_buffer` enabled there are two additional tests failures: `test_undefined_read_typed_buffer_as_untyped_dxbc`  and `test_bindless_uav_counter_sm51`